src/main/java/io/github/cottonmc/energy/impl/EnergyHandler.java

```diff
@@ -60,4 +61,9 @@ void setMaxEnergy(int amount) {
 		maxEnergy = amount;
 		if (currentEnergy > maxEnergy) currentEnergy = maxEnergy;
 	}
+
+	@Override
+	public void listen(@Nonnull Runnable r) {
```

Comment:
Okay but we always need to call this if we change something.

Reply:
What do we call it with? We'd need to supply a Runnable.

Reply:
Look at the Concrete classes. They all have an onChanged(), which calls the cached Runnable. Any method in which the Component changes, calls onChanged on itself